smithy-aws-iam-traits/src/main/java/software/amazon/smithy/aws/iam/traits/ActionNameTrait.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.smithy.aws.iam.traits;
+
+import software.amazon.smithy.model.FromSourceLocation;
+import software.amazon.smithy.model.shapes.ShapeId;
+import software.amazon.smithy.model.traits.StringTrait;
+
+public final class ActionNameTrait extends StringTrait {
+
+    public static final ShapeId ID = ShapeId.from("aws.iam#actionName");
+
+    private ActionNameTrait(String action, FromSourceLocation sourceLocation) {
+        super(ID, action, sourceLocation);
+    }
+
```

Comment:
Can this have an additional simple constructor that takes only the `action` and passes `SourceLocation.NONE`?

Reply:
added

```
    private ActionNameTrait(String action) {
        super(ID, action, SourceLocation.NONE);
    }
```